modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/PartitionPruneTest.java

```diff
@@ -0,0 +1,557 @@
+                assertNodes();
+
+                assertTrue(res.isEmpty());
+            }, new Object[]{ null });
+
+        execute("select * from T1 where T1.ID is NULL and T1.ID = ?",
+            res -> {
+                assertPartitions();
+                assertNodes();
+
+                assertTrue(res.isEmpty());
+            }, 123);
+
+        execute("select * from T1 where T1.ID = ? and T1.ID = ?",
+            res -> {
+                assertPartitions();
+                assertNodes();
+
+                assertTrue(res.isEmpty());
+            }, null, 123);
+
+        execute("select * from T1 where T1.ID is NULL or T1.ID = ?",
+            res -> {
+                assertPartitions(partition("T1_CACHE", 123));
+                assertNodes(node("T1_CACHE", 123));
+
+                assertEquals(1, res.size());
+                assertEquals("name_123", res.get(0).get(1));
+            }, 123);
+
+        execute("select * from T1 where T1.ID = ? or T1.ID = ?",
+            res -> {
+                assertPartitions(partition("T1_CACHE", 123));
+                assertNodes(node("T1_CACHE", 123));
+
+                assertEquals(1, res.size());
+                assertEquals("name_123", res.get(0).get(1));
+            }, null, 123);
+    }
+
+    /** */
+    @Test
+    public void testEmptyConditions() {
+        execute("select * from T1 where T1.ID = ? and T1.ID = ?",
+            res -> {
+                assertPartitions();
+                assertNodes();
+
+                assertTrue(res.isEmpty());
+            },
+            123, 518);
+
+        execute("select * from T1 where T1.ID = ? and (T1.ID = ? OR T1.ID = ? OR T1.ID = ?)",
+            res -> {
+                assertPartitions();
+                assertNodes();
+
+                assertTrue(res.isEmpty());
+            },
+            123, 518, 781, 295);
+
+        execute("select * from T1 where (T1.ID = ? OR T1.ID = ?) AND (T1.ID = ? OR T1.ID = ?)",
+            res -> {
+                assertPartitions();
+                assertNodes();
+
+                assertTrue(res.isEmpty());
+            },
+            123, 518, 781, 295);
+
+        execute("select * from T1 where (T1.ID = ? AND T1.ID = ?) OR (T1.ID = ? AND T1.ID = ?)",
+            res -> {
+                assertPartitions();
+                assertNodes();
+
+                assertTrue(res.isEmpty());
+            },
+            123, 518, 781, 295);
+    }
+
+    /** */
+    @Test
+    public void testSelectIn() {
+        IntStream.of(2, 6).forEach(i -> {
+            testSelect(i, true, "_KEY");
+            testSelect(i, true, "ID");
+        });
+    }
+
+
+    /** */
+    @Test
+    public void testSelectOr() {
+        testSelect(1, false, "_KEY");
+        testSelect(1, false, "ID");
+
+        IntStream.of(2, 6).forEach(i -> {
+            testSelect(i, false, "_KEY");
+            testSelect(i, false, "ID");
+        });
+    }
+
+    /** */
+    @Test
+    public void testSimpleJoin() {
```

Comment:
Let's add test with multiple imputs, but not join (union or other set op for example)